getflywheel/local-addon-boilerplate
# Modernize Boilerplate

## Summary

Some maintenance to get the "New Addon Boilerplate" up to a good starting point. The main changes:

- Upgrade Local core and Local components dependencies
- Provide additional typings to allow saving data to the Site model
- Move where the navigation item is added from the "More" tab to the "Tool" tab
- Use the updated `siteData` service from `LocalMain.getServiceContainer()`
- Add an example of creating a child logger instance

## QA

The [create-local-addon](https://github.com/getflywheel/create-local-addon) only pulls from released versions of this boilerplate repo, so to QA this PR, install this repo like you would any other addon:

1. Within the `addons` folder of the Local user data folder, clone this repo. On mac that would be `~/Library/Application Support/Local/addons`
2. Install the deps with `yarn install`
3. Build with `yarn build`
4. Start Local and enable the addon
5. Restart Local and ensure that the counter still saves to the site under "Tools > Counter"
